code/game/objects/structures/training_machine.dm

```diff
@@ -0,0 +1,234 @@
+	. = ..()
+	if (pinned_target)
+		vis_contents.Cut()
+		user.put_in_hands(pinned_target)
+		pinned_target = null
+		handle_density()
+		to_chat(user, "<span class='notice'>You remove the target from the seat.</span>")
+		playsound(src, "rustle", 50, TRUE)
+
+/obj/structure/training_machine/proc/toggle()
+	if (moving)
+		moving = FALSE
+	else
+		start_moving()
+
+/obj/structure/training_machine/proc/stop_moving(message = "Ending training simulation.")
+	moving = FALSE
+	starting_turf = null
+	say(message)
+	playsound(src,'sound/machines/synth_no.ogg',50,FALSE)
+
+/obj/structure/training_machine/proc/start_moving()
+	moving = TRUE
+	starting_turf = get_turf(src)
+	say("Beginning training simulation.")
+	playsound(src,'sound/machines/triple_beep.ogg',50,FALSE)
+	do_movement()
+
+/obj/structure/training_machine/proc/do_movement()
+	var/turf/current_turf = get_turf(src)
+	if (!moving || !starting_turf || isspaceturf(current_turf))
+		stop_moving()
+		return
+	if (current_turf == target_position) //We've reached our target turf, now find a new one
+		target_position = null
+	if (!target_position)
+		target_position = find_target_position()
+		if (!target_position)
+			stop_moving("ERROR! Cannot calculate suitable movement path.")
+	var/turf/nextStep = get_step_towards(src, target_position)
+	if (!Move(nextStep))
+		target_position = null //We couldn't move towards the target turf, so find a new target turf
+	if (obj_flags & EMAGGED && world.time > last_attack_time + attack_cooldown)
+		var/mob/living/carbon/target = locate() in view(1, src)
+		if (target?.stat == CONSCIOUS)
+			do_attack_animation(target, null, secret_evil_toolbox)
+			target.adjustBruteLoss(secret_evil_toolbox.force)
+			playsound(src,'sound/weapons/smash.ogg',50,FALSE)
+			last_attack_time = world.time
+
+	addtimer(CALLBACK(src, .proc/do_movement), MAX_SPEED - move_speed)
+
+/obj/structure/training_machine/proc/find_target_position()
+	var/list/turfs = list()
+	for(var/turf/potential_turf in view(range, starting_turf))
+		if (potential_turf.is_blocked_turf() || potential_turf == target_position)
+			continue
+		turfs += potential_turf
+	if (!length(turfs))
+		return
+	return pick(turfs)
+
+/obj/structure/training_machine/proc/handle_density()
+	if(length(buckled_mobs) || pinned_target || (obj_flags & EMAGGED))
+		density = TRUE
+	else
+		density = FALSE
+
+/obj/structure/training_machine/buckle_mob(mob/living/M, force = FALSE, check_loc = TRUE)
+	. = ..()
+	if (pinned_target)
+		return FALSE
+
+/obj/structure/training_machine/post_buckle_mob()
+	handle_density()
+	return ..()
+
+/obj/structure/training_machine/post_unbuckle_mob()
+	handle_density()
+	return ..()
+
+/obj/structure/training_machine/bullet_act(obj/projectile/P)
+	if(pinned_target)
+		pinned_target.bullet_act(P)
+	else
+		. = ..()
+
+/obj/structure/training_machine/emag_act(mob/user)
+	. = ..()
+	if (obj_flags & EMAGGED)
+		return
+	obj_flags |= EMAGGED
+	secret_evil_toolbox = new(src)
```

Comment:
is this toolbox dropped when the emagged training machine is destroyed? if so, is that intentional?

Reply:
maybe and if so yes
each one of these comes with a secret evil toolbox built in don't you know

Reply:
ok fine